lib/msf/core/payload/linux/aarch64/meterpreter_loader.rb

```diff
@@ -0,0 +1,53 @@
+#
+# In memory loader used to execute Mettle ELF file.
+# Compatible with Kernel Linux >= 3.17 (where memfd_create is introduced)
+# Author: Martin Sutovsky <martin_sutovsky[at]rapid7.com>
+# Resource and Credits: https://magisterquis.github.io/2018/03/31/in-memory-only-elf-execution.html
+#
+module Msf::Payload::Linux::Aarch64::MeterpreterLoader
+  def in_memory_load(payload)
+      in_memory_loader = [
```

Comment:
It would be really helpful to comment the stager asm code bit more- the syscall offsets are not universal across architectures, so saying what syscalls are being used in each would help.  It also looks like you're doing a couple things at once, which would be nice to call out.  It would be really useful to include labels or comments about where branches go as well.